packages/ethers-typescript-typings/index.d.ts

```diff
@@ -0,0 +1,29 @@
+declare module 'ethers-contracts' {
+    export interface TransactionDescription {
+        name: string;
+        signature: string;
+        sighash: string;
+        data: string;
+    }
+    export interface CallDescription extends TransactionDescription {
+        parse: (...args: any[]) => any;
+    }
+    export interface FunctionDescription {
+        (...params: any[]): TransactionDescription | CallDescription;
+        inputs: { names: string[]; types: string[] };
+        outputs: { names: string[]; types: string[] };
+    }
+    export interface EventDescription {
+        parse: (...args: any[]) => any;
+        inputs: { names: string[]; types: string[] };
+        signature: string;
+        topic: string;
+    }
+    // tslint:disable-next-line:max-classes-per-file
```

Comment:
Let's make this rule hold for this entire file. Unless you are sure we would never add another class.